tests/test_bookmarks.py

```diff
@@ -0,0 +1,143 @@
+                    self.assertIsNotNone(second_bookmark_value)
+                    
+                    self.assertIsNotNone(first_bookmark_value[replication_key])
+                    self.assertIsNotNone(second_bookmark_value[replication_key])
+
+                    # verify the 2nd bookmark is equal to 1st sync bookmark
+                    self.assertEqual(first_bookmark_value[replication_key],
+                                     second_bookmark_value[replication_key])
+
+                    for record in first_sync_messages:
+                        replication_key_value_utc = self.parse_date(record.get(replication_key))
+                        # verify 1st sync bookmark value is the max replication key value for a given stream
+                        self.assertLessEqual(replication_key_value_utc, first_bookmark_value_utc,
+                                             msg="First sync bookmark was set incorrectly, a record with a greater replication key value was synced")
+
+                    for record in second_sync_messages:
+                        replication_key_value_utc = self.parse_date(record.get(replication_key))
+                        # verify the 2nd sync replication key value is greater or equal to the 1st sync bookmarks
+                        self.assertGreaterEqual(replication_key_value_utc, simulated_bookmark_utc,
+                                                msg="Second sync records do not respect the previous bookmark")
+                        # verify the 2nd sync bookmark value is the max replication key value for a given stream
+                        self.assertLessEqual(replication_key_value_utc, second_bookmark_value_utc,
+                                             msg="Second sync bookmark was set incorrectly, a record with a greater replication key value was synced")
+
+                    # verify that we get less data in the 2nd sync
+                    self.assertLess(second_sync_count, first_sync_count,
+                                    msg="Second sync does not have less records, bookmark usage not verified")
+
+                elif expected_replication_method == self.FULL_TABLE:
+
+                    # verify the syncs do not set a bookmark for full table streams
+                    self.assertIsNone(first_bookmark_value)
+                    self.assertIsNone(second_bookmark_value)
+
+                    # verify the number of records in the second sync is the same as the first
+                    self.assertEqual(second_sync_count, first_sync_count)
+
+                else:
+                    raise NotImplementedError(
+                        "INVALID EXPECTATIONS\t\tSTREAM: {} REPLICATION_METHOD: {}".format(stream, expected_replication_method))
+
+                # Verify at least 1 record was replicated in the second sync
+                self.assertGreater(
+                    second_sync_count, 0, msg="We are not fully testing bookmarking for {}".format(stream))
```

Comment:
// all records match
for record in sync_1_records:
     self.assertIn(record, sync_2_records)

Reply:
Are you looking for [this](https://github.com/singer-io/tap-iterable/blob/e239e1fde6b1af7e951b09c15164c268acc69349/tests/test_bookmarks.py#L115-#L122)

Reply:
No this was for the FULL_TABLE streams.  Some small value add to make sure the record details are also in tact as well as the number of records.

Reply:
Fix is [here](https://github.com/singer-io/tap-iterable/blob/TDL-21942-add-tap-tester-tests/tests/test_bookmarks.py#L166-#L168)